src/connectors/MagicConnector.ts

```diff
@@ -0,0 +1,111 @@
+import { ChainId, ProviderType } from '@dcl/schemas'
+import { ConnectorUpdate } from '@web3-react/types'
+import { getConfiguration } from '../configuration'
+import { AbstractConnector } from './AbstractConnector'
+// tslint:disable-next-line
+import type { InstanceWithExtensions, SDKBase } from '@magic-sdk/provider'
+// tslint:disable-next-line
+import type { OAuthExtension } from '@magic-ext/oauth'
+
+export class MagicConnector extends AbstractConnector {
+  private chainId: ChainId
+  private account: string | null
+  private magic: InstanceWithExtensions<SDKBase, OAuthExtension[]> | undefined
+
+  constructor(desiredChainId: ChainId) {
+    super({ supportedChainIds: getConfiguration()[ProviderType.MAGIC].chains })
+    this.chainId = desiredChainId
+    this.account = null
+  }
+
+  activate = async (): Promise<ConnectorUpdate<string | number>> => {
+    this.magic = await this.buildMagicInstance(this.chainId)
+    const isLoggedIn = await this.magic.user.isLoggedIn()
```

Comment:
Other providers throw an error if they can't connect, shouldn't we do the same here?

Reply:
Changed it to throwing an error if the user isn't logged in.